src/script/lua_api/l_env.cpp

```diff
@@ -635,35 +635,46 @@ int ModApiEnvMod::l_find_nodes_in_area(lua_State *L)
 	INodeDefManager *ndef = getServer(L)->ndef();
 	v3s16 minp = read_v3s16(L, 1);
 	v3s16 maxp = read_v3s16(L, 2);
+	v3s16 cube = maxp - minp + 1;
+
+	/* Limit for too large areas, assume default values
+	 * and give tolerances of 1 node on each side
+	 * (chunksize * MAP_BLOCKSIZE + 2)^3 = 551368
+	*/
+	if (cube.X * cube.Y * cube.Z > 551368) {
+		luaL_error(L, "find_nodes_in_area(): area volume"
+				" exceeds allowed value of 551368");
+		return 0;
+	}
+
 	std::set<content_t> filter;
```

Comment:
Not possible. `INodeDefManager::getIds` (used below) only takes a "regular" set. Changing that is out of the scope of this PR.